### PR TITLE
Fix some __CUDACC__ vs __CUDA_ARCH__, add clarifying comment

### PR DESCRIPTION
**Pull request by lgritz**

### CODE REVIEW

File: src/include/OSL/platform.h [hunk 1]
```diff
@@ -114,6 +114,15 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #endif
 
 
+// Note about __CUDACC__ versus __CUDA_ARCH__:
+//
+// __CUDACC__     defined any time we are compiling a module for Cuda
+//                (both for the host pass and the device pass). "Do this
+//                when using nvcc or clang with ptx target."
+// __CUDA_ARCH__  is only defined when doing the host pass. "Do this only
```
Comment (ee33) [inline src/include/OSL/platform.h]: __CUDA_ARCH__: do you mean device pass?

Comment (lgritz) [reply src/include/OSL/platform.h]: Yes, that is what I meant. :-)

Comment (lgritz) [reply src/include/OSL/platform.h]: pushed the fix, thanks

**Comment by lgritz** [comment]
OK now?

**Comment by ee33** [comment]
LGTM!

**Pull request merged**